packages/contact-exchange/src/tests/MediatorCoordination.test.ts

```diff
@@ -0,0 +1,187 @@
+  let eventBus: EventEmitter;
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+    eventBus = new EventEmitter();
+    const securityService = new SecurityService();
+    service = new DidService(eventBus, securityService);
+    (uuidv4 as jest.Mock).mockReturnValue('test-uuid');
+  });
+
+  describe('processMediatorOOB', () => {
+    it('should throw an error for invalid OOB format', async () => {
+      await expect(service.processMediatorOOB('invalid-oob')).rejects.toThrow(
+        'Invalid OOB format. Missing encoded payload.',
+      );
+    });
+
+    it('should throw an error if the decoded OOB lacks "from" field', async () => {
+      const mockOob =
+        'oob=' + Buffer.from(JSON.stringify({})).toString('base64url');
+      await expect(service.processMediatorOOB(mockOob)).rejects.toThrow(
+        'Invalid OOB content. Missing "from" field.',
+      );
+    });
+
+    it('should successfully process valid OOB and handle keylist update', async () => {
+      const mockOob =
+        'oob=' +
+        Buffer.from(JSON.stringify({ from: 'did:peer:123' })).toString(
+          'base64url',
+        );
+
+      const mockDidPeerMethod = {
+        generateMethod2: jest.fn().mockResolvedValue({
+          did: 'did:peer:456',
+          privateKeyE: { id: 'key1' },
+          privateKeyV: { id: 'key2' },
+        }),
+        generateMethod2RoutingKey: jest.fn().mockResolvedValue({
+          did: 'did:example:mediatorNewDID',
+          privateKeyE: { id: 'key3' },
+          privateKeyV: { id: 'key4' },
+        }),
+      };
+      (DidPeerMethod as jest.Mock).mockImplementation(() => mockDidPeerMethod);
+
+      const mockResolver = {
+        resolve: jest.fn().mockResolvedValue({
+          service: [
+            {
+              type: 'DIDCommMessaging',
+              serviceEndpoint: { uri: 'http://test.com' },
+            },
+          ],
+        }),
+      };
+      (PeerDIDResolver as jest.Mock).mockImplementation(() => mockResolver);
+
+      (fetch as jest.Mock)
+        .mockResolvedValueOnce({
+          ok: true,
+          json: jest.fn().mockResolvedValue({
+            '@type':
+              'https://didcomm.org/coordinate-mediation/2.0/mediate-grant',
+            body: {
+              message_type:
+                'https://didcomm.org/coordinate-mediation/2.0/mediate-grant',
+              routing_did: 'did:example:mediatorRoutingKey',
+              from: 'did:example:mediatorNewDID',
+            },
+          }),
+        })
+        .mockResolvedValueOnce({
+          ok: true,
+          json: jest.fn().mockResolvedValue({
+            type: 'https://didcomm.org/coordinate-mediation/2.0/keylist-update-response',
+            body: {
+              updated: [
+                {
+                  recipient_did: 'did:example:mediatorNewDID',
+                  action: 'add',
+                  result: 'success',
+                },
+              ],
+            },
+          }),
+        });
```

Comment:
Could you give an explanation for why the values you struggle to return from these mocks are not even used subsequently?

```suggestion
      (fetch as jest.Mock)
        .mockResolvedValue({
          ok: true,
          json: jest.fn().mockResolvedValue({}),
        });
```

Reply:
I mistakenly pushed this, but it was adjusted in the last 2 commit, if you check well.

Reply:
Nothing changed regarding this.

Reply:
Oh sorry i thought of something else, they mocked values for the Mediator Response and KeylistUpdate are used, after executing the test you could see it from the logs.

Reply:
Apply my suggestion, and check if the test still passes. Note that you should also avoid logging sensitive information. On my side, it still works and I want to know why.

```ts
(fetch as jest.Mock)
    .mockResolvedValue({
      ok: true,
      json: jest.fn().mockResolvedValue({}),
    });
```

Reply:
Thanks for addressing this, i agree the test will run without those values.